homeassistant/components/stream/hls.py

```diff
@@ -1,13 +1,21 @@
 """Provide functionality to stream HLS."""
-import io
+from copy import copy
 
 from aiohttp import web
 
 from homeassistant.core import callback
 
-from .const import FORMAT_CONTENT_TYPE, MAX_SEGMENTS, NUM_PLAYLIST_SEGMENTS
+from .const import (
+    FORMAT_CONTENT_TYPE,
+    MAX_SEGMENTS,
+    NUM_PLAYLIST_SEGMENTS,
+    PART_DURATION,
+    STREAM_TIMEOUT,
+)
 from .core import PROVIDERS, HomeAssistant, IdleTimer, StreamOutput, StreamView
-from .fmp4utils import get_codec_string, get_init, get_m4s
+from .fmp4utils import get_codec_string
+
+ADVANCE_PART_LIMIT = max(3 / PART_DURATION, 3)
```

Comment:
Sure, it's straight from the RFC 6.2.5.2:

>    If the _HLS_msn is greater than the Media Sequence Number of the last
>    Media Segment in the current Playlist plus two, or if the _HLS_part
>    exceeds the last Partial Segment in the current Playlist by the
>    Advance Part Limit, then the server SHOULD immediately return Bad
>    Request, such as HTTP 400.  The Advance Part Limit is three divided
>    by the Part Target Duration if the Part Target Duration is less than
>    one second, or three otherwise.